### PR TITLE
First version of a Basic Angular app

### PR DESCRIPTION
**Pull request by aullman**
This is using Angular 5. Still need to get the Unit and E2E tests to pass.

### CODE REVIEW

File: Angular-Basic-Video-Chat/src/app/opentok.service.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  connectSession() {
+    if (config.API_KEY && config.TOKEN && config.SESSION_ID) {
+      return this.connect(config.API_KEY, config.SESSION_ID, config.TOKEN);
+    } else {
+      return fetch(config.SAMPLE_SERVER_BASE_URL + '/session')
+        .then((data) => data.json())
+        .then((json) => this.connect(json.apiKey, json.sessionId, json.token));
+    }
```
Comment (msach22) [inline Angular-Basic-Video-Chat/src/app/opentok.service.ts]: @aullman In case there's an error, can we have a  catch the error handle it appropriately?

Comment (aullman) [reply Angular-Basic-Video-Chat/src/app/opentok.service.ts]: Yeah, I am handling it in the app.component.ts. It's not giving a very good error message though, I'll fix that.

Comment (aullman) [reply Angular-Basic-Video-Chat/src/app/opentok.service.ts]: I noticed though we're not handling errors in the Basic Video Chat either. We should do that. Right now if you don't populate the config.js file you just get

```
Failed to load resource: net::ERR_NAME_NOT_RESOLVED
```

Comment (msach22) [reply Angular-Basic-Video-Chat/src/app/opentok.service.ts]: @aullman will add that to the list! Thanks!

**Pull request merged**